extension/js/common/api/email-provider/gmail/gmail.ts

```diff
@@ -277,14 +278,20 @@ export class Gmail extends EmailProviderApi implements EmailProviderInterface {
     if (format === 'full') {
```

Comment:
@martgil the changes only involve messages loaded in `full` format.
I guess, the clause below `} else { // format === raw` should behave similarly.
Please don't copy/paste code, extract common processing statements into a helper function.

Reply:
ok sir - I looked into the block of code inside of ` else { // format === raw` and don't find a reliable way to "detect" for a bogus message so I had to rely on different error message to check based on `result.error.message`.

I then add a conditional checks for the error message `This message / key probably does not conform to a valid OpenPGP format.` - this gives me a closest way to detect potential plain text inside of pgp header/footer

<img width="619" alt="Screen Shot 2022-03-16 at 6 20 07 PM" src="https://user-images.githubusercontent.com/46025304/158570064-974fca83-9727-48d0-b98c-5447eb89174d.png">

Reply:
> isn't this the place where we could add the .search(...) condition?

Sorry, I run out of an idea of what to search for?

I started thinking if I can add a condition in the `} else { // raw` and can't find a suitable condition checks since the `armoredMsg` always returned a good looking armored message eg. `Header + content + Footer` wherein the content seems not suitable for checking so I went to rely on a specific error - in this case the format error that says the message/key does not conform to a valid pgp format.

Moreover, here is what's happening with the given scenario:

assume that there was a message:
```
----BEGIN PGP MESSAGE-----
Version: FlowCrypt Email Encryption 8.2.6
Comment: Seamlessly send and receive encrypted email

wV4DgCPkBPUgCC4SAQdAK4uPXBhuG9iaUDoXs3LqKrq2AHjEByD3F7UhqrCj
aaslakjdlaksjdlkajsdlkajsdlkjaslkjaslkdjalksdjlaksdjlaksjlkasdlaskdlakjsdlkasadklasd
```

this message will fall under the condition where `(format === 'full')` and will show:
![image](https://user-images.githubusercontent.com/46025304/158972159-7f53cbac-5ff1-4479-a51d-2309ea65f3ce.png)

the following message will fall under the `else` condition `(raw)`
```
-----BEGIN PGP MESSAGE-----
Version: FlowCrypt Email Encryption 8.2.6
Comment: Seamlessly send and receive encrypted email

wV4DgCPkBPUgCC4SAQdAK4uPXBhuG9iaUDoXs3LqKrq2AHjEByD3F7UhqrCj
aaslakjdlaksjdlkajsdlkajsdlkjaslkjaslkdjalksdjlaksdjlaksjlkasdlaskdlakjsdlkasadklasd

-----END PGP MESSAGE-----
```

I would add a test and add further changes if needed.

Reply:
I set up a (failing) unit test to test Gmail.extractArmoredBlock directly and added a comment.
Please have a look.

Reply:
thanks sir, @rrrooommmaaa - What I did is replaced the `throw FormatError...` within the `raw === format` else codeblock and return the plaintext instead.